cmd/kubeadm/app/cmd/phases/reset/preflight.go

```diff
@@ -51,8 +53,14 @@ func runPreflight(c workflow.RunData) error {
 		return errors.New("preflight phase invoked with an invalid data struct")
 	}
 
-	if !r.ForceReset() && !r.DryRun() {
-		klog.Warning("[reset] WARNING: Changes made to this host by 'kubeadm init' or 'kubeadm join' will be reverted.")
+	if (!r.ForceReset() && !r.DryRun()) || r.CleanupTmpDir() {
```

Comment:
i think we can just remove handling of the cleanup temp dir flag here and printing the warning below. the flag is opt in, so users know the passed it.